src/core/materialized-view.ts

```diff
@@ -7,18 +7,24 @@ import { createCachedAccumulator } from './pipeline/accumulators';
 import { evaluateExpression, toHashExpression } from './pipeline/expressions';
 import type { Expression } from './pipeline/expressions';
 export interface MaterializedViewDefinition {
+  db?: string;
+  collection?: string;
   groupBy: Expression;
   accumulatorDefs: AccumulatorDefinition[];
 }
 
 export class MaterializedView implements CollectionListener {
+  private _db: string | undefined;
+  private _collection: string | undefined;
```

Comment:
From your own comment 
>db and collection are stored in MV because they are necessary for findEligibleMV

Why can it be undefined then ?

Reply:
They are not always useful in the tests already written on the Materialized View: but this is not a sufficient reason :)

I removed the undefined type and forced their use in all tests.